projects/angular-svg-icon/src/lib/svg-icon.component.ts

```diff
@@ -174,41 +157,28 @@ export class SvgIconComponent implements OnInit, OnDestroy, OnChanges, DoCheck {
 
 			this.copyNgContentAttribute(elem, icon);
 
-			if (this.klass && this.applyClass) {
-				this.setClass(elem.firstChild, null, this.klass);
-			}
-
-			if (this.svgClass) {
-				this.setClass(elem.firstChild, null, this.svgClass);
-			}
+			this.svg.update(x => x + 1);
+		}
+	}
 
-			if (this.viewBox) {
-				if (this.viewBox === 'auto') {
-					// Attempt to convert height & width to a viewBox.
-					const w = icon.getAttribute('width');
-					const h = icon.getAttribute('height');
-					if (h && w) {
-						const vb = `0 0 ${w} ${h}`;
-						this.renderer.setAttribute(icon, 'viewBox', vb);
-						this.renderer.removeAttribute(icon, 'width');
-						this.renderer.removeAttribute(icon, 'height');
-					}
-				} else if (this.viewBox !== '') {
-					this.renderer.setAttribute(icon, 'viewBox', this.viewBox);
+	private updateViewBox(viewBox?: string) {
+		if (viewBox) {
+			const icon = this.elemSvg;
+			if (viewBox === 'auto') {
+				// Attempt to convert height & width to a viewBox.
+				const w = icon.getAttribute('width');
+				const h = icon.getAttribute('height');
+				if (h && w) {
+					const vb = `0 0 ${w} ${h}`;
+					this.renderer.setAttribute(icon, 'viewBox', vb);
 					this.renderer.removeAttribute(icon, 'width');
 					this.renderer.removeAttribute(icon, 'height');
 				}
+			} else if (viewBox !== '') {
+				this.renderer.setAttribute(icon, 'viewBox', viewBox!);
+				this.renderer.removeAttribute(icon, 'width');
+				this.renderer.removeAttribute(icon, 'height');
 			}
-
-			this.stylize();
-
-			// If there is not a svgAriaLabel and the SVG has an arial-label, then do not override
-			// the SVG's aria-label.
-			if (!(this.svgAriaLabel === undefined && elem.firstChild.hasAttribute('aria-label'))) {
-				this.doAria(this.svgAriaLabel || '');
-			}
```

Comment:
What is the behavior now with aria labels?

Reply:
This changes has been moved inside doAria method.